### PR TITLE
[new] Add spritejs via single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by lizheming**
# Pull request template for new library

Pull request for issue: #12844 
Related issue(s): #12844 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/spritejs/spritejs
* Official website (optional, not the repository): https://spritejs.org
* NPM package url (optional): https://www.npmjs.com/package/spritejs
* License and its reference: MIT
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 81
  * Count of stars: 3095
  * Count of forks: 182
* NPM download stats (optional):
  * Downloads in the last day: 93
  * Downloads in the last week: 654
  * Downloads in the last month: 3318

## Essential checklist

* [ ] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [ ] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [ ] The parent of the commit(s) in the PR is not older than 3 days.
* [ ] Pull request is sent from a non-master branch with a meaningful name.
* [ ] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [ ] Close corresponding issue in commit message
* [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/spritejs/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "version": "3.2.8",
+  "description": "A cross platform high-performance graphics system.",
+  "filename": "spritejs.js",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/spritejs/spritejs.git"
+  },
+  "author": "akira-cn",
+  "homepage": "http://spritejs.org",
+  "keywords": [],
```
Comment (xtuc) [inline ajax/libs/spritejs/package.json]: keywords can be used for the search on the website, you can add the one spritejs uses

Comment (lizheming) [reply ajax/libs/spritejs/package.json]: OK, I taken the topic words into it which show in the github repo https://github.com/spritejs/spritejs, please check again, thank you.

File: ajax/libs/spritejs/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "spritejs",
+  "version": "3.2.8",
```
Comment (MattIPv4) [inline ajax/libs/spritejs/package.json]:
```suggestion
```
Please do not include the version attr, the bot will automatically provide this when it pulls in files.

File: ajax/libs/spritejs/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "A cross platform high-performance graphics system.",
+  "filename": "spritejs.js",
```
Comment (MattIPv4) [inline ajax/libs/spritejs/package.json]:
```suggestion
  "filename": "spritejs.min.js",
```
As a minified file is available in the package, we should suggest it by defualt

File: ajax/libs/spritejs/package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  "author": "akira-cn",
+  "homepage": "http://spritejs.org",
+  "keywords": ["spritejs", "canvas", "graphics", "rendering", "render", "svg-path", "svg", "webgl2", "webgl", "3d", "2d"],
+  "license": "MIT",
+  "npmName": "spritejs",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "spritejs.js",
```
Comment (MattIPv4) [inline ajax/libs/spritejs/package.json]: Are these the only files needed, not the `es` files?

```
├── dist
│   ├── spritejs.es.js
│   ├── spritejs.es.min.js
│   ├── spritejs.js
│   ├── spritejs.min.js
│   ├── spritejs.worker.es.js
│   ├── spritejs.worker.es.min.js
│   ├── spritejs.worker.js
│   └── spritejs.worker.min.js
```

Comment (lizheming) [reply ajax/libs/spritejs/package.json]: yes you're right, I should add es module file into it. And I asked author `spritejs.worker.js` is a bundle for web worker, it also should be added. Thank you for your figure out, I fixed it and please check again.

File: ajax/libs/spritejs/package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        "spritejs.min.js",
+        "spritejs.es.js",
+        "spritejs.es.min.js",
+        "spritejs.worker.js",
+        "spritejs.worker.min.js",
+        "spritejs.worker.es.js",
+        "spritejs.worker.es.min.js"
```
Comment (MattIPv4) [inline ajax/libs/spritejs/package.json]:
```suggestion
        "spritejs.*"
```
This is loading every spritejs file from dist, so we can replace this with a single-line glob.

Comment (lizheming) [reply ajax/libs/spritejs/package.json]: wow~ that's awesome, I didn't know we can write like this, thanks~

**Pull request merged**